### ISSUE TITLE
eval: `--effort` cannot be set, so an A/B cannot reproduce a benchmark that pins one

### ISSUE THREAD

**Issue by vlad-ryzhkov** [zernie/agent-lint#189]
Read on `origin/main` at `85c54e7`; latest published is `vigiles@24.0.0`.

`paid_runEval` can pick the model but not the effort level, so an A/B cannot be run at the effort
a published benchmark pins — and the arm you do run is not the arm the author measured.

`claude` has had the flag for a while:

```console
$ claude --help | grep -A1 'effort <level>'
  --effort <level>                      Effort level for the current session
                                        (low, medium, high, xhigh, max)
```

The driver builds the argv itself and never passes it (`src/eval.ts:382-402`):

```ts
    const args = [
      "-p",
      a.task,
      "--output-format",
      "stream-json",
      "--verbose",
      "--model",
      a.model,
      "--permission-mode",
      "acceptEdits",
      ...(a.pluginDir !== undefined
        ? ["--plugin-dir", resolve(a.pluginDir)]
        : []),
      ...(a.hasSettings ? ["--settings", "settings.json"] : []),
      "--allowedTools",
      ...a.tools,
    ];
```

`grep -nE '"--effort"' src/` returns nothing on `origin/main`. `src/harness-test.ts:460` builds a
second argv with the same gap.

## Why it blocks a real measurement

`docs/measuring-skills.md:24` sets the bar:

> The harness is loaded as it ships — the real `claude` CLI, the skill injected exactly as a user
> installs it.

Effort is part of how the harness runs. A skill author who publishes numbers at one effort cannot
be checked at that effort.

Concrete case. I was A/B-ing a proposed patch to the `benjamin-plus` skill
(`JetBrains/benjamin-plus-skill`). Its results file pins the setting in `EXPECTED-RESULTS.md:125`:

> Claude Code pinned 2.1.201 both arms; `anthropic/claude-sonnet-5`; effort low.

vigiles runs that model at the default effort instead, so my arms and the author's headline are not
the same configuration. Effort also plausibly moves the thing under test here — the skill is a
context-cost ruleset, and reasoning budget is a large share of a session's tokens.

## What I had to do instead

A shim first on `PATH` that re-execs the real binary with the flag:

```bash
#!/usr/bin/env bash
REAL="${BP_REAL_CLAUDE:-/path/to/claude}"
if [ -n "${BP_EFFORT:-}" ]; then
  exec "$REAL" --effort "$BP_EFFORT" "$@"
fi
exec "$REAL" "$@"
```

Verified it reaches the binary by passing a bad value and getting `claude`'s own reply:

```
Warning: Unknown --effort value 'bogus' — ignoring it and using the default effort.
Valid values: low, medium, high, xhigh, max.
```

It works, and it is the wrong shape: nothing in the run record says which effort produced the
numbers, and it silently defeats the lock — see below.

## The part that is a correctness bug, not a feature gap

If effort is added to the argv only, the eval lock will replay across effort levels.

`EvalLockInputs` (`src/eval-lock.ts:86-102`) carries `model`, `evalApiVersion`, and an opaque
`inputs`, and its own doc says the hash must cover everything

> that steer[s] the model

(`src/eval-lock.ts:97-99`). Effort steers the model. Filled at `src/eval.ts:1721-1725`:

```ts
  const inputsHash = evalInputsHash({
    model: args.model,
    evalApiVersion: lock.evalApiVersion,
    inputs: args.inputs,
  });
```

So today a `--check` or a replay would hand back a result recorded at a different effort with no
warning. My `PATH` shim has exactly that hole right now.

## Suggested change

1. `AgentRunArgs` (`src/eval.ts:334`) gains `readonly effort?: string`; the argv gets the same shape
   already used for `--plugin-dir`:

   ```ts
   ...(a.effort !== undefined ? ["--effort", a.effort] : []),
   ```

2. The specs that already carry `model?: string` gain a sibling `effort?: string`
   (`src/eval.ts:133`, `:186`, `:523`, `:658`, `:2022`), threaded where the model is defaulted
   (`:596`, `:685`, `:1833`). Same addition in `src/harness-test.ts:460`.

3. `EvalLockInputs` gains `readonly effort?: string` next to `model`, filled at `src/eval.ts:1721`.
   `src/eval-lock.test.ts:46` — "evalInputsHash changes when ANY model-affecting field changes" — is
   where the assertion belongs.

Omitting an effort keeps today's behaviour byte-identical, so nothing existing has to change.

Happy to send a PR if the shape above is what you want.

**Comment by zernie** [comment]
Shipped in #191 (`4f0e6e8`). Your shape was right in structure and I changed it in three places — here is each, so a divergence does not read as a misread.

**What shipped.** `effort` on the five spec surfaces beside `model`, on the argv, and hashed. It is also an arm in its own right, so `arms: { cheap: { effort: "low" }, rich: { effort: "high" } }` runs through the same significance machinery model-as-an-arm already uses. The lock and the reports record which effort produced the numbers — the half of your complaint your `PATH` shim could not fix.

**Measured.** `npm run check` 18/18; 3462 unit tests. Eight mutations, each proven to have landed.

---

## Three things done differently

**1. Not a literal union — `string | number` plus a runtime refusal.**

Your issue quotes the valid set as `low, medium, high, xhigh, max`, which is what a union would encode. Three facts said otherwise. The parser is case-insensitive, keeps an alias map, and accepts an integer budget — the binary's own agent-file validation says `Valid options: … or an integer`. And that set has already **moved**: the 2.1.42 bundle has `["low","medium","high","max"]` with zero occurrences of `xhigh`; 2.1.257 has all five. A union would have rejected a valid level the next time upstream adds one.

The silent fallback a union was meant to prevent is not silent, and your own transcript is the evidence — the binary prints `Unknown --effort value 'bogus' — ignoring it…` to stderr, and `spawnAgent` already captures stderr for rate-limit detection. So a rejected value is caught at runtime and **throws**, rather than being counted as a `runError` trial. A `runError` is dropped from the denominator, which is right for a transient and wrong here: a bad effort value is deterministic, so every trial would fail it and the report would be a rate computed over zero samples.

**2. `src/harness-test.ts:460` was not changed.**

You named it as a second site with the same gap. That tier drives a scripted mock, and its argv hard-codes `"claude-sonnet-4-5"` — `model` is not threaded there at all, and effort only shapes a request body the mock does not capture. Adding the flag would have been dead argv, so it is left alone deliberately rather than by oversight.

**3. `audit` gets no `--effort` flag.**

The audit probe shares `runSkillSelectionTrial` with `measureSelectionMatrix`, so the option exists on the shared type. It is exposed for the assertable test primitive and not as an `audit` CLI flag: `audit` is a local report, not a reproducibility surface, and a flag nobody can act on is surface without a use.

---

## Two things your report did not name, both of which would have made the fix wrong

**The env var beats the flag.** Effort has three inputs — `--effort`, the `effortLevel` settings key, and `CLAUDE_CODE_EFFORT_LEVEL` (12 references in 2.1.257) — and the env var sits above the flag. `EPHEMERAL_ALLOW_PREFIXES` passes `CLAUDE_*` through by design, so an ambient value survives even the *scrubbed* ephemeral env. Implementing points 1–3 alone would therefore have produced exactly the failure you filed: a lock recording `effort: "low"` over a run that executed at the shell's `max`, with more confidence than before because the field now exists. So effort is **pinned** — declared sets the var, and **omitted deletes an inherited one**, because "omit" has to mean the harness default rather than whatever the machine exported.

**The local cache has the same hole as the lock.** You named `EvalLockInputs`. `CacheKeyInput` in `src/eval-cache.ts` carries `model` with no effort in exactly the same way; fixing only the lock would have left a replay path open. Both are keyed now.

---

## Worth knowing before you rely on it

- **`arm.settings: { effortLevel: "low" }` already worked** for `runEval`/`measure` before this change, and is hashed in both the cache and the lock — a correct interim, unlike the `PATH` shim, which defeats the lock. It does not reach the trigger tier (`hasSettings: false` there) and sits below the env var, which is why the first-class field is still the right answer.
- **Codex refuses a declared effort** rather than dropping it. Its config has a `model_reasoning_effort` key, but nothing here has driven the real binary through it, and recording a budget the run did not use is worse than refusing.
- **"Omitted" is not a fixed level.** The default is per-model and can move between builds, so an effort-less lock is reproducible only modulo that — the same class of caveat as the harness version, and stated in the docs rather than implied.

Thank you for this one. The report that lands is the one that separates the feature gap from the correctness bug, and yours did — the lock paragraph is why the fix covers both hashes instead of just the flag.

---
_Generated by [Claude Code](https://claude.ai/code)_
